.test-infra/jenkins/Flink.groovy

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import DockerPublisher
+
+class Flink {
+  private static final String flinkVersion = '1.7'
+  private static final String flinkDownloadUrl = 'https://archive.apache.org/dist/flink/flink-1.7.0/flink-1.7.0-bin-hadoop28-scala_2.11.tgz'
+  private static final String FLINK_DIR = '"$WORKSPACE/src/.test-infra/dataproc"'
+  private static final String FLINK_SCRIPT = 'flink_cluster.sh'
+  private def job
+  private String jobName
+  private String jobServerImageTag
+
+  Flink(job, String jobName) {
+    this.job = job
+    this.jobName = jobName
+    this.jobServerImageTag = ''
+  }
+
+  /**
+   * Creates Flink cluster and specifies cleanup steps.
+   *
+   * @param sdkHarnessImages - the list of published SDK Harness images tags
+   * @param workerCount - the initial number of worker nodes
+   * @param slotsPerTaskmanager - the number of slots per Flink task manager
+   */
+  void setUp(List<String> sdkHarnessImages, Integer workerCount, Integer slotsPerTaskmanager = 1) {
+    setupFlinkCluster(sdkHarnessImages, workerCount, slotsPerTaskmanager)
+    addTeardownFlinkStep()
+  }
```

Comment:
Can you place `setupFlinkCluster` method below `setup()`?

Reply:
I'd rather keep it as is. I think public methods should be on top to avoid scrolling down implementation details when looking at the interface.

Reply:
It seems there is not one convention in Beam's codebase. I always preferred this: https://www.oracle.com/technetwork/java/codeconventions-141855.html (I know it's Java, but Groovy is java's "younger brother" ;) ) 

> These methods should be grouped by functionality rather than by scope or accessibility. For example, a private class method can be in between two public instance methods. The goal is to make reading and understanding the code easier.

The main idea behind such ordering is that whenever you read a method, you start from the most generic parts and then (if needed) sink into the details, which are right below (private methods). Other than that, most of Java/Groovy codebases I worked with was organized this way.

Right now if I want to dig into the details (private methods) I have to scroll through all public methods to find them. 

> I think public methods should be on top to avoid scrolling down implementation details when looking at the interface.

I think that whenever I want to see the whole interface, I can look it up by pressing ctrl + space and see the list of methods that are available.